config/routes.rb

```diff
@@ -17,6 +17,7 @@
       member do
         get 'refresh'
         post 'refresh'
+        get 'courses.json', to: 'courses#show_json', format: 'json', as: 'one_course_json'
```

Comment:
Good catch. That `.json` is not needed.

That mapping to `/courses/:id/course.json` is because netbeans tmc-plugin adds `/courses.json` to request. Now student gets to select only one course.

Your earlier suggestion (magic url) is then part of little bit bigger changes where enrolment is coming to play.